client/src/components/Search.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { Component } from "react";
+import ListCourse from "./ListCourse";
+
+class Search extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { initial: this.props.items, current: this.props.items };
+  }
+
+  filterItems(input) {
+    var updatedlist = this.state.initial;
+    updatedlist = updatedlist.filter(item => {
+      return item.code.search(input.target.value) !== -1;
+    });
+    this.setState({ current: updatedlist });
+  }
+
+  render() {
+    return (
+      <div>
+        <input type="text" onChange={this.filterItems.bind(this)} />
```

Comment:
this `onChange` line doesn't seem right to me

Reply:
why though? Its syntatically correct and works fine!

Reply:
> Just because you're correct doesn't mean you're right

Reply:
@Aviral14 Its syntactically correct, but can be inefficient. 
You're creating a new binding every time the state of the textfield changes. If these bindings were with a large number of GUI elements for example, it could cause problems. 

Binding it once in the constructor is good enough. :)

Reply:
Exactly what I wanted to say.

Reply:
@Kushagra-0801  bc to kaha kyun nahi
All this shows how paranoid u r!!